### ISSUE TITLE
24bit 48Khz Noise

### ISSUE THREAD

**Issue by vanboom** [fabnicol/dvda-author#2]
I am having the same issue as described in this post:
https://sourceforge.net/p/dvd-audio/mailman/message/917873/

Certain tracks exhibit white noise, while others play perfectly.  I have tried various DVD-RW and DVD-R medium and drives to rule out the burn process.  Does the odd/even samples issue still exist? 

.flac files are my source material.   thanks!

**Comment by fabnicol** [comment]
Thanks for reporting.
I've never been able to replicate such issues.
The odd sample issue might be an idea, perhaps there was a regression
somewhere in the (protracted) process of implementing multichannel audio
processing.
On the other hand the fact that you have flac files, not wav files, as
input does not seem to go along with that assumption.
OK, if your audio is not too confidential please send me a personal mail
with references to audio files uploaded to dropbox or an equivalent
platform so that I can try and fix things.
Fabrice Nicol

2018-05-02 23:51 GMT+02:00 Don <notifications@github.com>:

**Comment by vanboom** [comment]
Thank you.  I have sent a personal mail with my source material.  Please let me know if you have any advice on where to look in the software.  I am happy to contribute if I can get up the learning curve on the source code.

**Comment by vanboom** [comment]
More info:  I have tried the various padding options and so far no success.

**Comment by fabnicol** [comment]
They have been deactivated of late.

**Comment by fabnicol** [comment]
PM did not get through.
Please resend links using a secure messaging service. I'm on Signal (Fabrice
Nicol).

**Comment by vanboom** [comment]
I have run some tests and this file seems to be one of the offenders. 

http://69.164.201.157/instrumental.flac   

DVD Layout:
`1     10   48000   24   2      8226653   ./instrumental.flac`

Notice the length is an odd number.  

To duplicate the test:  
1. rip a CD and convert to 24bit/48khz.   I use the Reaper DAW to do this. 
1.I have a script that will gather a directory full of .flac files and build the `image.iso`.

````
#!/bin/bash
dvda-author -P0 -o DVD -g ./*.wav  -F
mkisofs -o image.iso -udf DVD
echo 'go ahead and write the iso to the DVD using standard tools'
````

**Info from my testing...**
* I ripped the Coldplay Parachutes album this way, and created a perfect DVD-Audio disk.  When I added the `instrumental.flac` track, the noise appeared on the second track. 

Notice the length of each ripped track is an multiple of 20, where my custom track has an odd length. 

````
DVD Layout
Group  Track    Rate Bits  Ch        Length  Filename

    1     01   48000   24   2      8226653   ./instrumental.flac
    1     02   48000   24   2      6634240   ./parachutes-001.flac
    1     03   48000   24   2     14449920   ./parachutes-002.flac
    1     04   48000   24   2     15349120   ./parachutes-003.flac
    1     05   48000   24   2     10948480   ./parachutes-004.flac
    1     06   48000   24   2     12986240   ./parachutes-005.flac
    1     07   48000   24   2     13051520   ./parachutes-006.flac
    1     08   48000   24   2      2265600   ./parachutes-007.flac
    1     09   48000   24   2     12257280   ./parachutes-008.flac
    1     10   48000   24   2     12019200   ./parachutes-009.flac
````
I hope this helps...I apologize I have not been able to get Signal working and my PM have been blocked.  

UPDATE:  dvda-author 9.0.3 is definitely rounding or padding...everything has an even Length.
 ````
   1     01   48000   24   2      8226652  ./instrumental.flac
```` 
8226652 vs. 8226653

**Comment by fabnicol** [comment]
Thanks Don I'll have a look at this.

-------- Message d'origine --------
De : Don <notifications@github.com> 
Date : 12/05/2018  01:56  (GMT+01:00) 
À : fabnicol/dvda-author <dvda-author@noreply.github.com> 
Cc : fabnicol <fabrnicol@gmail.com>, Comment <comment@noreply.github.com> 
Objet : Re: [fabnicol/dvda-author] 24bit 48Khz Noise (#2) 

I have run some tests and this file seems to be one of the offenders.
http://69.164.201.157/instrumental.flac
DVD Layout:

1 10 48000 24 2 8226653 ./instrumental.flac
Notice the length is an odd number.
To duplicate the test:

rip a CD and convert to 24bit/48khz.   I use the Reaper DAW to do this.

1.I have a script that will gather a directory full of .flac files and build the image.iso.

#!/bin/bash
dvda-author -P0 -o DVD -g ./*.wav  -F
mkisofs -o image.iso -udf DVD
echo 'go ahead and write the iso to the DVD using standard tools'

Info from my testing...

I ripped the Coldplay Parachutes album this way, and created a perfect DVD-Audio disk.  When I added the instrumental.flac track, the noise appeared on the second track.

Notice the length of each ripped track is an multiple of 10, where my custom track has an odd length.
DVD Layout
Group  Track    Rate Bits  Ch        Length  Filename

    1     01   48000   24   2      8226653   ./instrumental.flac
    1     02   48000   24   2      6634240   ./parachutes-001.flac
    1     03   48000   24   2     14449920   ./parachutes-002.flac
    1     04   48000   24   2     15349120   ./parachutes-003.flac
    1     05   48000   24   2     10948480   ./parachutes-004.flac
    1     06   48000   24   2     12986240   ./parachutes-005.flac
    1     07   48000   24   2     13051520   ./parachutes-006.flac
    1     08   48000   24   2      2265600   ./parachutes-007.flac
    1     09   48000   24   2     12257280   ./parachutes-008.flac
    1     10   48000   24   2     12019200   ./parachutes-009.flac

I hope this helps...I apologize I have not been able to get Signal working and my PM have been blocked.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/fabnicol/dvda-author","title":"fabnicol/dvda-author","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/fabnicol/dvda-author"}},"updates":{"snippets":[{"icon":"PERSON","message":"@vanboom in #2: I have run some tests and this file seems to be one of the offenders. \r\n\r\nhttp://69.164.201.157/instrumental.flac   \r\n\r\nDVD Layout:\r\n`1     10   48000   24   2      8226653   ./instrumental.flac`\r\n\r\nNotice the length is an odd number.  \r\n\r\nTo duplicate the test:  \r\n1. rip a CD and convert to 24bit/48khz.   I use the Reaper DAW to do this. \r\n1.I have a script that will gather a directory full of .flac files and build the `image.iso`.\r\n\r\n````\r\n#!/bin/bash\r\ndvda-author -P0 -o DVD -g ./*.wav  -F\r\nmkisofs -o image.iso -udf DVD\r\necho 'go ahead and write the iso to the DVD using standard tools'\r\n````\r\n\r\n**Info from my testing...**\r\n* I ripped the Coldplay Parachutes album this way, and created a perfect DVD-Audio disk.  When I added the `instrumental.flac` track, the noise appeared on the second track. \r\n\r\nNotice the length of each ripped track is an multiple of 10, where my custom track has an odd length. \r\n\r\n````\r\nDVD Layout\r\nGroup  Track    Rate Bits  Ch        Length  Filename\r\n\r\n    1     01   48000   24   2      8226653   ./instrumental.flac\r\n    1     02   48000   24   2      6634240   ./parachutes-001.flac\r\n    1     03   48000   24   2     14449920   ./parachutes-002.flac\r\n    1     04   48000   24   2     15349120   ./parachutes-003.flac\r\n    1     05   48000   24   2     10948480   ./parachutes-004.flac\r\n    1     06   48000   24   2     12986240   ./parachutes-005.flac\r\n    1     07   48000   24   2     13051520   ./parachutes-006.flac\r\n    1     08   48000   24   2      2265600   ./parachutes-007.flac\r\n    1     09   48000   24   2     12257280   ./parachutes-008.flac\r\n    1     10   48000   24   2     12019200   ./parachutes-009.flac\r\n````\r\nI hope this helps...I apologize I have not been able to get Signal working and my PM have been blocked.  "}],"action":{"name":"View Issue","url":"https://github.com/fabnicol/dvda-author/issues/2#issuecomment-388513098"}}}
{"@type":"MessageCard","@context":"http://schema.org/extensions","hideOriginalBody":"false","originator":"37567f93-e2a7-4e2a-ad37-a9160fc62647","title":"Re: [fabnicol/dvda-author] 24bit 48Khz Noise (#2)","sections":[{"text":"","activityTitle":"**Don**","activityImage":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","activitySubtitle":"@vanboom","facts":[]}],"potentialAction":[{"name":"Add a comment","@type":"ActionCard","inputs":[{"isMultiLine":true,"@type":"TextInput","id":"IssueComment","isRequired":false}],"actions":[{"name":"Comment","@type":"HttpPOST","target":"https://api.github.com","body":"{\"commandName\":\"IssueComment\",\"repositoryFullName\":\"fabnicol/dvda-author\",\"issueId\":2,\"IssueComment\":\"{{IssueComment.value}}\"}"}]},{"name":"Close issue","@type":"HttpPOST","target":"https://api.github.com","body":"{\"commandName\":\"IssueClose\",\"repositoryFullName\":\"fabnicol/dvda-author\",\"issueId\":2}"},{"targets":[{"os":"default","uri":"https://github.com/fabnicol/dvda-author/issues/2#issuecomment-388513098"}],"@type":"OpenUri","name":"View on GitHub"},{"name":"Unsubscribe","@type":"HttpPOST","target":"https://api.github.com","body":"{\"commandName\":\"MuteNotification\",\"threadId\":331406374}"}],"themeColor":"26292E"}

**Comment by fabnicol** [comment]
Hem, well.
My test was OK with your flac file.
The thing is you forgot to add the `-dvd-audio` option to the `mkisofs` command line
Il you have a recent version of Jörg Schilling's original `cdrtools` packages (mkisofs+cdrecord), then this option is available. Note: this is NOT true of the lousy `woodim` package to be found in many GNU/Linux distros. Check `cdrecord -version` to see if this is the right thing. Otherwise rebuild cdrecord from original source after uninstalling the faulty one.
This done, incidentally, you can either use the `--mkisofs` or the `--cdrecord` options to avoid scripting:

`dvda-author-dev -g instrumental.flac a2_24_44.wav --cdrecord`

---------- DVD-A author version 2016-dev ----------

[LICENSE]  Copyright Dave Chapman 2005-Fabrice Nicol 2007-2016
<fabnicol@users.sourceforge.net>-Lee and Tim Feldkamp 2008-2009
This file is part of dvda-author.
dvda-author is free software: you can redistribute it and/or modify 
it under the terms of the GNU General Public License as published by 
the Free Software Foundation, either version 3 of the License, or 
(at your option) any later version.
dvda-author is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of 
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
GNU General Public License for more details.
You should have received a copy of the GNU General Public License 
along with dvda-author.  If not, see http://www.gnu.org/licenses/.


Current time is: 13 May 2018, 23h 11m 31s

------------------------------------------------------------
[PAR]  Log file is: log.txt
[PAR]  allocate_files = 1
[PAR]  Make ISO image then run cdrecord to burn disc image.
[WAR]  Device command  will be interpolated.
       Run cdrecord -scanbus to check for available drivers
[INF]  Found extensible WAV header
[INF]  Found `fact' chunk
[MSG]  Proceeding with same file...

------------------------------------------------------------

DVD Layout
Group  Track    Rate Bits  Ch        Length  Filename

    1     01   48000   24   2      8226653   instrumental.flac
    1     02   44100   24   2       441000   a2_24_44.wav


[MSG]  Size of raw PCM data: 52005918 bytes (49.60  MB)
[INF]  Processing instrumental.flac
[INF]  Writing first packet - pack=0, bytesinbuffer=65532
[INF]  Writing last packet - pack=24630, bytesinbuffer=1422
[INF]  Processing a2_24_44.wav
[INF]  Writing first packet - pack=0, bytesinbuffer=65532
[INF]  Writing last packet - pack=1320, bytesinbuffer=744
[INF]  Creating /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output/AUDIO_TS/ATS_01_0.IFO
[MSG]  /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output/AUDIO_TS/ATS_01_0.IFO was created.
[INF]  Creating /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output/AUDIO_TS/ATS_01_0.BUP
[MSG]  /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output/AUDIO_TS/ATS_01_0.BUP was created.
[MSG]  SAMG pointers
       Last audio group=0
       Last audio track=0
       Absolute sector pointer to last AOB sector=26710
[INF]  Creating /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output/AUDIO_TS/AUDIO_PP.IFO
[MSG]  /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output/AUDIO_TS/AUDIO_PP.IFO was created.
[MSG]  Total size of AUDIO_TS: 26026 sectors
[MSG]  Start offset of  VIDEO_TS in ISO file: 26713 sectors,  offset 54708224

[INF]  Creating /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output/AUDIO_TS/AUDIO_TS.IFO
[MSG]  /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output/AUDIO_TS/AUDIO_TS.IFO was created.
[INF]  Creating /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output/AUDIO_TS/AUDIO_TS.BUP
[MSG]  /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output/AUDIO_TS/AUDIO_TS.BUP was created.


Group   Title  Track  First Sect   Last Sect  First PTS  PTS length cga

    1   01/02     1           0       24630         585    15424974   1
    1   02/02     2       24631       25951         585      900000   1

Total number of tracks: 2.

[INF]  Launching: /usr/bin/mkisofs to create image
[DBG]  application:/usr/bin/mkisofs args1:-dvd-audio
Setting input-charset to 'UTF-8' from locale.
3.02a09 (x86_64-pc-linux-gnu)
Scanning /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output
Scanning /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output/VIDEO_TS
Scanning /home/fab/Dev/dvda-author-dev/build/linux/.dvda-author/temp/output/AUDIO_TS
Writing:   Initial Padblock                        Start Block 0
Done with: Initial Padblock                        Block(s)    16
Writing:   Primary Volume Descriptor               Start Block 16
Done with: Primary Volume Descriptor               Block(s)    1
Writing:   End Volume Descriptor                   Start Block 17
Done with: End Volume Descriptor                   Block(s)    1
Writing:   UDF volume recognition area             Start Block 18
Done with: UDF volume recognition area             Block(s)    3
Writing:   Version block                           Start Block 21
Done with: Version block                           Block(s)    1
Writing:   UDF pad to sector 32                    Start Block 22
Done with: UDF pad to sector 32                    Block(s)    10
Writing:   UDF main seq                            Start Block 32
Done with: UDF main seq                            Block(s)    16
Writing:   UDF second seq                          Start Block 48
Done with: UDF second seq                          Block(s)    16
Writing:   UDF integ seq                           Start Block 64
Done with: UDF integ seq                           Block(s)    2
Writing:   UDF pad to sector 256                   Start Block 66
Done with: UDF pad to sector 256                   Block(s)    190
Writing:   UDF Anchor volume                       Start Block 256
Done with: UDF Anchor volume                       Block(s)    1
Writing:   UDF file set                            Start Block 257
Done with: UDF file set                            Block(s)    2
Writing:   UDF directory tree                      Start Block 259
Done with: UDF directory tree                      Block(s)    6
Writing:   UDF file entries                        Start Block 265
Done with: UDF file entries                        Block(s)    6
Writing:   Path table                              Start Block 271
Done with: Path table                              Block(s)    4
Writing:   Directory tree                          Start Block 275
Done with: Directory tree                          Block(s)    3
Writing:   Directory tree cleanup                  Start Block 278
Done with: Directory tree cleanup                  Block(s)    0
Writing:   The File(s)                             Start Block 278
 18,98% done, estimate finish Sun May 13 23:11:31 2018
 37,85% done, estimate finish Sun May 13 23:11:31 2018
 56,72% done, estimate finish Sun May 13 23:11:31 2018
 75,71% done, estimate finish Sun May 13 23:11:31 2018
 94,58% done, estimate finish Sun May 13 23:11:31 2018
Total translation table size: 0
Total rockridge attributes bytes: 0
Total directory bytes: 4096
Path table size(bytes): 42
Done with: The File(s)                             Block(s)    26026
Writing:   UDF Anchor end volume                   Start Block 26304
Done with: UDF Anchor end volume                   Block(s)    1
Writing:   UDF Pad end                             Start Block 26305
Done with: UDF Pad end                             Block(s)    150
Max brk space used 0
26455 extents written (51 MB)
[MSG]  Image was created with size 52910 KB.

[INF]  Launching cdrecord to burn disc
[DBG]  application:/usr/bin/cdrecord args1:-s
cdrecord: No write mode specified.
Cdrecord-ProDVD-ProBD-Clone 3.02a09 (x86_64-pc-linux-gnu) Copyright (C) 1995-2016 Joerg Schilling
cdrecord: Assuming -sao mode.
cdrecord: If your drive does not accept -sao, try -tao.
cdrecord: Future versions of cdrecord may have different drive dependent defaults.
cdrecord: Insufficient 'file read' privileges. You will not be able to open all needed devices.
cdrecord: Insufficient 'file write' privileges. You will not be able to open all needed devices.
cdrecord: Insufficient 'device' privileges. You may not be able to send all needed SCSI commands, this my cause various unexplainable problems.
cdrecord: Insufficient 'memlock' privileges. You may get buffer underruns.
Using libscg version 'schily-0.9'.
cdrecord: Insufficient 'priocntl' privileges. You may get buffer underruns.
cdrecord: Insufficient 'network' privileges. You will not be able to do remote SCSI.
Linux sg driver version: 3.5.36
No target specified, trying to find one...
Using dev=6,0,0.
Device type    : Removable CD-ROM
Version        : 5
Response Format: 2
Capabilities   : 
Vendor_info    : 'HL-DT-ST'
Identifikation : 'BDDVDRW CH10LS28'
Revision       : '1.00'
Device seems to be: Generic mmc2 DVD-R/DVD-RW/DVD-RAM.
Using generic SCSI-3/mmc-3 DVD+RW driver (mmc_dvdplusrw).
Driver flags   : NO-CD DVD MMC-3 SWABAUDIO BURNFREE 
Supported modes: PACKET SAO LAYER_JUMP
Starting to write CD/DVD/BD at speed 4 in real SAO mode for single session.
Last chance to quit, starting real write    0 seconds. Operation starts.
Running pad based emulation to blank the medium.
secsize 2048 padbytes 2048000 padblocks 1000 maxblocks 2295104
cdrecord: Cannot blank DVD+RW media with non Ricoh based drive.
Track 00:    0 of    1 MB pad written.
Track 02: Total bytes read/written: 54179840/54179840 (26455 sectors).
[MSG]  User execution time: 0 minutes 0 seconds 535541 microseconds
[MSG]  System execution time: 0 minutes 0 seconds 64309 microseconds

**Comment by vanboom** [comment]
Oh my, I have been burning dvd-audio for YEARS without that flag.  I will give it a try tonight and report back.  Thanks!

**Comment by vanboom** [comment]
Sorry Fabrice, I am still getting noise when using the --dvd-audio for mkisofs. 

Here are my source files - everything needed to reproduce the issue: http://69.164.201.157/dvdatest.tar.gz

I build the `image.iso` using `genflac.sh` script which runs `dvda-author-dev` and `mkisofs`.  
My version of `mkisofs` is the latest from Cdrtools 3.0.1, dvda-author-dev built from source. 
Thanks for  any further help or advice.

**Comment by fabnicol** [comment]
OK Don I'll try to replicate.

2018-05-17 1:33 GMT+02:00 Don <notifications@github.com>:

**Comment by fabnicol** [comment]
OK Don, thanks very much for reporting.
I could replicate this time.
The flac'ing is not at issue, it's another cause.
I'll see to it and will come back to you when fixed.

**Comment by fabnicol** [comment]
Hi Don
Thanks for reporting on this issue, as it made clear that I had underestimated one important thing dating back to Lee Feldkamp's 2009 contribution. 
This chunk of code dealt with the case in which the number of audio samples is not a multiple of a DVD-AUDIO-related parameter which varies according to number of channels and bit depth (`sampleunitsize` in the code).
As the 2ch/24bit/48kHz audio files have sampleunitsizes of 12 bytes, and the samples are only 6 bytes, and odd number of samples causes a violation of sampleunitsize constraints in the audio buffers, resulting in a 6-byte shift in LPCM audio (AOB files), which suffices to cause an audio mess.
These issues have been corrected now I think. However it remains to be seen whether the fix works for multichannel too.
I uploaded a fresh linux build on sourceforge. It is a libgcc-static build so it should work on a variety of recent GNU/Linux platforms.
